laalaa/apps/advisers/models.py

```diff
@@ -156,14 +157,49 @@ def __iter__(self):
 
 
 IMPORT_STATUSES = Choices(
-    ("RUNNING", "running"), ("SUCCESS", "success"), ("FAILURE", "failure"), ("ABORTED", "aborted")
+    ("CREATED", "created"),
+    ("RUNNING", "running"),
+    ("SUCCESS", "success"),
+    ("FAILURE", "failure"),
+    ("ABORTED", "aborted"),
 )
 
 
+class ImportManager(models.Manager):
+    @staticmethod
+    def abort_last():
+        last_import = Import.objects.last()
```

Comment:
Sorry, I said Manager but maybe I meant `Queryset` class. Basically, I like the idea of this being chainable. I think you can define a queryset subclass, use `self` for `Import.objects`, and then have Django generate you a `Manager` class from that `Queryset` class (and I think it should also make the method available on the `Manager`, they brought that in at some point I think)

Reply:
Or it can probably stay on this and I wouldn't hate it, but I think there might be a way of writing the `Import.objects.last()` to be based on `self` or something.

Reason being, and it absolutely won't come up on this but it's good to get it fundamentally right, is that if this manager were to be used (or subclassed and used) other than as the default manager, it would be using a manager other than itself to get the `last()` from. Now, that's unlikely, and in the case of `last()` really not important because realistically both managers are going to have the same implementation of `last()`, inherited from the same ancestor.

But still, I could write a bug built on this type of behaviour if I chose to, so I think it's best avoided.